frontend/src/components/Ingredients/Ingredients.tsx

```diff
@@ -1,19 +1,91 @@
-import { useEffect, useState } from 'react'
-import { fetchAllIngredients } from '../apiCalls'
-import './Ingredients.css'
+import { Ingredient } from '../apiCalls';
 
+interface Props {
+  ingredients: Array<Ingredient>;
+  setIngredients: Function;
+}
 
-export function Ingredients() {
+export function Ingredients({ ingredients, setIngredients }: Props) {
 
-  const [ingredients, setIngredients] = useState([])
-  
-  useEffect(() => {
-    fetchAllIngredients().then(data => setIngredients(data))
-  }, [])
+  function sortIngredients(criteria: keyof Ingredient) {
+    const sortedIngredients = [...ingredients].sort((a, b) => {
+      const aAttribute = a[criteria]
+      const bAttribute = b[criteria]
+      if(typeof aAttribute === "number" && typeof bAttribute === "number") {
+        return aAttribute - bAttribute
+      }
+      return (aAttribute as string).localeCompare(bAttribute as string)
+    });
+    setIngredients(sortedIngredients)
+  }
+
+  const renderedIngredients = ingredients.map((ingredient) => {
+    return (
+      <tr key={ingredient.id}>
+        <td className="border border-slate-700 px-1">{ingredient.name}</td>
+        <td className="border border-slate-700 px-1">
+          {parseFloat(ingredient.protein_in_grams).toFixed(1)}g
```

Comment:
It'd be nice to DRY this up
`{parseFloat(ingredient.protein_in_grams).toFixed(1)}g`